src/main/java/com/uid2/shared/model/Site.java

```diff
@@ -113,6 +146,7 @@ public Site(int id, String name, String description, Boolean enabled, Set<Client
 
     public void setEnabled(Boolean enabled) { this.enabled = enabled; }
     public void setDomainNames(Set<String> domainNames) { this.domainNames = domainNames; }
+    public void setAppNames(Set<String> appNames) { this.appNames = appNames; }
```

Comment:
This allows appNames to be null, but the ctor doesn't alllow it. 

Maybe we can make this setter do 
`this.appNames = (appNames != null) ? appNames : new HashSet<>();`
and then the constructor calls this setter instead?

Reply:
I agree it is good to avoid the duplicate code, but after some further consideration and I got some new ideas. Maybe it is good to keep the format as assigning the value directly in the constructor?

1. Clarity and Intent: Constructors are typically used to set initial values directly, making it clear and explicit what values are being initialized when an object is created. Calling setter methods within a constructor could potentially confuse readers of the code about where initializations are happening.
2. Format: It ensures consistency in the constructor format, streamlining it to one single line of code which is not complicated.

Reply:
It's not complicated to duplicate this code, but it did lead to this bug / inconsistency (allowing null in the setter but not in the constructor). I think it makes sense to DRY if the setter does anything more than `this.x = x`

Reply:
Sure, I will make the change. I feel both way work for me. I do not have a strong preference.

Reply:
One way to avoid this inconsistency of ctor vs Setter is to use the `@Nonnull` annotation on the field. Lombok will add an explicit Null Check in the Setter method.

> A number of annotations from popular libraries that indicate non-nullness, such as javax.annotation.Nonnull, if present on the field, result in an explicit null check in the generated setter.

https://projectlombok.org/features/GetterSetter
The responsibility now falls on the caller to not set a null value for the field